src/Service/Ajaxcom.php

```diff
@@ -90,7 +92,10 @@ public function handle(string $view, array $parameters = []): JsonResponse
         $ajax = $this->replaceClass->handle($ajax);
         $ajax = $this->replaceJavaScripts->handle($ajax, $view, $parameters);
         $ajax = $this->callbacks->handle($ajax);
-        $ajax = $this->changeUrl->handle($ajax);
+
+        if (false === $this->doNotChangeUrl) {
```

Comment:
I don't like this `if`.

You should make change without changing this method eg. you can change internal state of `changeUrl` handler and let it decide if the url should be changed or not.

Do this, please in same way as other handlers works eg. within `doNotChangeUrl()` method change some internal state of the handler itself.